lib/resolve-value.js

```diff
@@ -27,26 +29,39 @@ function toString(value) {
 var resolveValue = function(decl, map, /*optional*/ignorePseudoScope, /*internal debugging*/_debugIsInternal) {
 	var debugIndent = _debugIsInternal ? '\t' : '';
 
+	var matchingVarDecl = undefined;
+	var RE_VAR_FUNC_G = new RegExp(RE_VAR_FUNC.source, 'g');
 	var resultantValue = toString(decl.value);
 	var warnings = [];
+	
 
-	var variablesUsedInValueMap = {};
-	// Use `replace` as a loop to go over all occurrences with the `g` flag
-	resultantValue.replace(new RegExp(RE_VAR_FUNC.source, 'g'), function(match, variableName, fallback) {
-		variablesUsedInValueMap[variableName] = true;
-	});
-	var variablesUsedInValue = Object.keys(variablesUsedInValueMap);
+	// match all variables first, to gather all available variables in decl.value
+	var variablesUsedInValue = [];
+	while ((matchingVarDecl = RE_VAR_FUNC_G.exec(resultantValue))) {
+		variablesUsedInValue.push(matchingVarDecl[1]);
+	}
+	// remove duplicates from array
+	variablesUsedInValue = variablesUsedInValue.filter(filterDistinct);
 
 	//console.log(debugIndent, (_debugIsInternal ? '' : 'Try resolving'), generateScopeList(decl.parent, true), `ignorePseudoScope=${ignorePseudoScope}`, '------------------------');
 
 	// Resolve any var(...) substitutons
 	var isResultantValueUndefined = false;
-	resultantValue = resultantValue.replace(new RegExp(RE_VAR_FUNC.source, 'g'), function(match, variableName, fallback) {
-		// Loop through the list of declarations for that value and find the one that best matches
-		// By best match, we mean, the variable actually applies. Criteria:
-		//		- is under the same scope
-		//		- The latest defined `!important` if any
-		var matchingVarDeclMapItem;
+	
+	// var() = var( <custom-property-name> [, <any-value> ]? )
+	// matches `name[, fallback]`, captures "name" and "fallback"
+	// See: http://dev.w3.org/csswg/css-variables/#funcdef-var
+	while ((matchingVarDecl = balanced('var(', ')', resultantValue))) {
+		var matchingVarDeclMapItem = undefined;
+
+		// Split at the comma to find variable name and fallback value
+		// There may be other commas in the values so this isn't necessary just 2 pieces
```

Comment:
*necessarily

Spelled it wrong in my comment before, https://github.com/MadLittleMods/postcss-css-variables/pull/97#discussion_r282411060

Reply:
fix'd